app/models/version.rb

```diff
@@ -4,6 +4,7 @@ class Version < ApplicationRecord
   belongs_to :rubygem, touch: true
   has_many :dependencies, -> { order("rubygems.name ASC").includes(:rubygem) }, dependent: :destroy, inverse_of: "version"
   has_one :gem_download, proc { |m| where(rubygem_id: m.rubygem_id) }, inverse_of: :version
+  has_one :joined_gem_downloads, class_name: "GemDownload"
```

Comment:
OMG! All this time I was thinking of ways to accommodate `where(rubygem_id: m.rubygem_id)`, without realizing we can just drop it :sob: 
`gem_downloads` `belongs_to` `rubygems` relation is redundant and was added to allow use of `GemDownload.find_by(rubygem_id: id, version_id: 0)` (downloads per gem). [Same was pointed out](https://github.com/rubygems/rubygems.org/pull/1222#discussion_r56884587)  when this is introduced and polymorphism was suggested. Two fully normalized tables would have avoided it as well.
Point being, this is a legendary fix :clap:  and we don't need to add a new `has_one`. We should update the existing `has_one :gem_download` on line 6 by dropping the `where(rubygem_id: m.rubygem_id)`.